File: SDKs/highwaysengland.co.uk/v1/python/src/sdk/sites.py
import requests
from typing import Optional
from sdk.models import shared, operations
from . import utils

class Sites:
    _client: requests.Session
    _security_client: requests.Session
    _server_url: str
    _language: str
    _sdk_version: str
    _gen_version: str

    def __init__(self, client: requests.Session, security_client: requests.Session, server_url: str, language: str, sdk_version: str, gen_version: str) -> None:
        self._client = client
        self._security_client = security_client
        self._server_url = server_url
        self._language = language
        self._sdk_version = sdk_version
        self._gen_version = gen_version

    
    def sites_index(self, request: operations.SitesIndexRequest) -> operations.SitesIndexResponse:
        r"""Get a list of sites
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/v{version}/sites", request.path_params)
        
        
        client = self._client
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.SitesIndexResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[shared.SiteResponse])
                res.site_response = out
        elif r.status_code == 400:
            pass
        elif r.status_code == 500:
            pass

        return res

    
    def get_v_version_sites_site_ids_(self, request: operations.GetVVersionSitesSiteIdsRequest) -> operations.GetVVersionSitesSiteIdsResponse:
        r"""Get selected sites
        """
        
        base_url = self._server_url
        
        url = utils.generate_url(base_url, "/v{version}/sites/{site_Ids}", request.path_params)
        
        
        client = self._client
        
        r = client.request("GET", url)
        content_type = r.headers.get("Content-Type")

        res = operations.GetVVersionSitesSiteIdsResponse(status_code=r.status_code, content_type=content_type)
        
        if r.status_code == 200:
            if utils.match_content_type(content_type, "application/json"):
                out = utils.unmarshal_json(r.text, Optional[shared.SiteResponse])
                res.site_response = out
        elif r.status_code == 400:
            pass
        elif r.status_code == 500:
            pass

        return res

    

File: SDKs/highwaysengland.co.uk/v1/typescript/src/sdk/sites.ts
import { AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";
import * as operations from "./models/operations";
import * as utils from "../internal/utils";

export class Sites {
  _defaultClient: AxiosInstance;
  _securityClient: AxiosInstance;
  _serverURL: string;
  _language: string;
  _sdkVersion: string;
  _genVersion: string;

  constructor(defaultClient: AxiosInstance, securityClient: AxiosInstance, serverURL: string, language: string, sdkVersion: string, genVersion: string) {
    this._defaultClient = defaultClient;
    this._securityClient = securityClient;
    this._serverURL = serverURL;
    this._language = language;
    this._sdkVersion = sdkVersion;
    this._genVersion = genVersion;
  }
  
  /**
   * sitesIndex - Get a list of sites
  **/
  sitesIndex(
    req: operations.SitesIndexRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.SitesIndexResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.SitesIndexRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/v{version}/sites", req.pathParams);
    
    const client: AxiosInstance = this._defaultClient!;
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.SitesIndexResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.siteResponse = httpRes?.data;
            }
            break;
          case httpRes?.status == 400:
            break;
          case httpRes?.status == 500:
            break;
        }

        return res;
      })
  }

  
  /**
   * getVVersionSitesSiteIds - Get selected sites
  **/
  getVVersionSitesSiteIds(
    req: operations.GetVVersionSitesSiteIdsRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.GetVVersionSitesSiteIdsResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.GetVVersionSitesSiteIdsRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/v{version}/sites/{site_Ids}", req.pathParams);
    
    const client: AxiosInstance = this._defaultClient!;
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.GetVVersionSitesSiteIdsResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.siteResponse = httpRes?.data;
            }
            break;
          case httpRes?.status == 400:
            break;
          case httpRes?.status == 500:
            break;
        }

        return res;
      })
  }

}
